src/Razor/src/Microsoft.AspNetCore.Razor.Language/TagHelperDescriptorBuilder.cs

```diff
@@ -52,6 +52,8 @@ public static TagHelperDescriptorBuilder Create(string kind, string name, string
 
         public abstract string TagOutputHint { get; set; }
 
+        public abstract bool CaseSensitive { get; set; }
```

Comment:
I think there's probably too many levels of case sensitivity here - or at least more than we know we need. What would you think about applying case sensitivity at the top level (tag helper).

Reply:
I had the same initial reaction. However, looking through our code base if we were to remove the CaseSensitive properties on the descriptors we'd need to flow that information independently through a good portion of our APIs. For instance, as one example take a look at the `TagHelperMatchingConventions` and all of the independent methods that inspect just the `BoundAttributeDescriptor` or just the `TagMatchingRuleDescriptor`. 

The core requirement of this feature is three pronged, in regards to enforcing case sensitivity. We need to enforce it for:
1. Required attributes
2. Bound Attributes
3. Required Tag Names

Given those three requirements we need to handle them in some fashion whether we pass additional "case sensitivity = true" parameters or have them be on the descriptors. Oh and, we could nuke the property on `AllowedChildDescriptor` since that concept doesn't exist in Component world.

I don't love what's here currently but I think @ajaybhargavb had to pick between the better of two evils. Let's use our Razor sync up today to see if we can come up with something better or agree on what's here.

Reply:
@NTaylorMullen described exactly why I took this approach.
In addition,
- I realize `AllowedChildDescriptor` doesn't make sense for components. But I did it for consistency and we won't be able to change it once we ship 3.0. So I just wanted to make sure we had this capability.
- Another advantage of adding `CaseSensitive` on individual required attributes/bound attribute/tag name etc is that it allows us to customize the descriptors to make it so that some are case sensitive while others aren't. I know we don't really need this now but I see no harm in having that.

> Let's use our Razor sync up today to see if we can come up with something better or agree on what's here.

👍